### PR TITLE
HSEC-2023-0015: cabal-install uses expired key policy

### PR DESCRIPTION
**Pull request by frasertweedale**
Add advisory for cabal-install key policy expiration check issue.

Heads-up @gbaz @Kleidukos - please review the advisory content and let us know if it needs tweaking.

## Advisory

- [ ] It's not duplicated
- [ ] All fields are filled
- [ ] It is validated by `hsec-tools`

### CODE REVIEW

**Comment by frasertweedale** [comment]
@blackheaven the CI skipped the check-advisories workflow, even though the `advsiories_changed`  step returned `should_skip=false`.  See https://github.com/haskell/security-advisories/actions/runs/6765685647.

Need to investigate why this happened...

**Review by Kleidukos — APPROVED** [review]
LGTM

**Comment by blackheaven** [comment]
I'll have a look

File: advisories/hackage/cabal-install/HSEC-2023-0015.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+ * Mirrors of Hackage cannot change the contents of packages. This
+   prevents the insertion of malicious code.
+
+ * Mirrors cannot omit newer packages for more than a few days
+   without clients noticing. This ensures both that mirrors cannot
+   maliciously deny security updates, and that mistakes in their
+   configuration will be noticed.
+
+Hackage has a [key policy file][] that delegates authority to a
```
Comment (blackheaven) [inline advisories/hackage/cabal-install/HSEC-2023-0015.md:50]: I'm not sure it matters, but `[The Update Framework]` does not have an extra `[]`

Comment (Kleidukos) [reply advisories/hackage/cabal-install/HSEC-2023-0015.md:50]: This is a fully lawful reference to https://github.com/haskell/security-advisories/pull/124/files#diff-67c9759c390b69f4c15040d1757b78e0cd35bc86900cd3e2c62347a1e069f338R80

Comment (frasertweedale) [reply advisories/hackage/cabal-install/HSEC-2023-0015.md:50]: @blackheaven I didn't intend it (I prefer the explicit trailing `[]` - a [*collapsed reference link*](https://spec.commonmark.org/0.30/#collapsed-reference-link) in commonmark jargon), but both are legal constructions.

File: .github/workflows/check-advisories.yml [hunk 2]
```diff
@@ -52,8 +39,8 @@ jobs:
           echo "code-hash=$code_hash" >> "$GITHUB_OUTPUT"
   check_advisories:
     name: Invoke check-advisories workflow
-    needs: [tools_changed, advisories_changed, code_hash]
-    if: ${{ needs.tools_changed.outputs.should_skip == 'true' && needs.advisories_changed.outputs.should_skip != 'true' }}
```
Comment (blackheaven) [inline .github/workflows/check-advisories.yml:56]: That's work duplication as advisories are checked whenever `hsec-tools` change

**Review by blackheaven — CHANGES_REQUESTED** [review]

**Comment by gbaz** [comment]
Looks good.

File: .github/workflows/check-advisories.yml (hunk 2 above)
Comment (frasertweedale) [reply .github/workflows/check-advisories.yml]: I removed this commit.

**Comment by frasertweedale** [comment]
CI (check-advisories workflow) passed: https://github.com/haskell/security-advisories/actions/runs/6779765269/job/18427358112.  Merging.

**Pull request merged**